sos/plugins/kubernetes.py

```diff
@@ -15,29 +15,38 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-from sos.plugins import Plugin, RedHatPlugin
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin
 from os import path
+from os.path import expanduser
 
 
-class kubernetes(Plugin, RedHatPlugin):
+class kubernetes(Plugin, RedHatPlugin, UbuntuPlugin):
 
     """Kubernetes plugin
     """
 
     # Red Hat Atomic Platform and OpenShift Enterprise use the
     # atomic-openshift-master package to provide kubernetes
     packages = ('kubernetes', 'kubernetes-master', 'atomic-openshift-master')
-    files = ("/etc/origin/master/master-config.yaml",)
+    files = ("/etc/origin/master/master-config.yaml")
 
+    # For IBM Cloud Private(ICP), run the sosreport like:
+    # sosreport -o kubernetes -k kubernetes.icp=True
     option_list = [
         ("all", "also collect all namespaces output separately",
             'slow', False),
         ("describe", "capture descriptions of all kube resources",
             'fast', False),
+        ("kubeconfig", "kubecofig file other than the default location",
+            'fast', ""),
+        ("icp", "set this option true if your installation is ICP based",
+            'fast', False),
         ("podlogs", "capture logs for pods", 'slow', False),
     ]
 
     def check_is_master(self):
+        if self.get_option('icp'):
+            return True
```

Comment:
ICP can never be master? This short-circuits the rest of this method call entirely when the `kubernetes.icp` option is given.

is that really what you intend?